libs/nest/aws/src/lib/aws.service.ts

```diff
@@ -1,24 +1,83 @@
 import { Injectable } from '@nestjs/common'
-import { S3 } from 'aws-sdk'
+import {
+  S3Client,
+  GetObjectCommand,
+  PutObjectCommand,
+  HeadObjectCommand,
+  DeleteObjectCommand,
+  GetObjectCommandOutput,
+} from '@aws-sdk/client-s3'
+import AmazonS3Uri from 'amazon-s3-uri'
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner'
+
+const MINUTE = 60
+const HOUR = 60 * MINUTE
+export type Encoding = 'binary' | 'base64' | 'utf-8'
 
 @Injectable()
 export class AwsService {
-  s3: S3
+  s3Client: S3Client
 
   constructor() {
-    this.s3 = new S3()
+    this.s3Client = new S3Client({})
   }
 
+  async getFile(fileName: string): Promise<GetObjectCommandOutput>
   async getFile(
     bucket: string,
     fileName: string,
-  ): Promise<AWS.S3.GetObjectOutput> {
-    const downloadParams = {
+  ): Promise<GetObjectCommandOutput>
+  async getFile(
+    bucket: string,
+    fileName?: string,
+  ): Promise<GetObjectCommandOutput> {
+    // If bucket is the only parameter, then it must be an S3 bucket URI
+    if (!fileName) {
+      // `bucket` is actually an S3 bucket URI here
+      const uri = AmazonS3Uri(bucket)
+      bucket = uri.bucket
+      fileName = uri.key
+    }
+
+    const command = new GetObjectCommand({
       Bucket: bucket,
       Key: fileName,
-    }
+    })
+
+    return await this.s3Client.send(command)
+  }
+
+  async getFileEncoded(
+    fileName: string,
+    encoding?: Encoding,
+  ): Promise<string | undefined>
+  async getFileEncoded(
+    bucket: string,
+    fileName: string,
+    encoding?: Encoding,
+  ): Promise<string | undefined>
+  async getFileEncoded(
+    bucket: string,
+    fileName?: string,
+    encoding: Encoding = 'base64',
+  ): Promise<string | undefined> {
+    const out = fileName
+      ? await this.getFile(bucket, fileName)
+      : await this.getFile(bucket)
+    return out.Body?.transformToString(encoding)
+  }
 
-    return await this.s3.getObject(downloadParams).promise()
+  async getFileB64(fileName: string): Promise<string | undefined>
+  async getFileB64(
+    bucket: string,
+    fileName: string,
+  ): Promise<string | undefined>
+  async getFileB64(
```

Comment:
~No need to minify B64
```suggestion
  async getFileBase64(fileName: string): Promise<string | undefined>
  async getFileBase64(
    bucket: string,
    fileName: string,
  ): Promise<string | undefined>
  async getFileBase64(
```

Reply:
Changed to getFileBase64